library/src/main/java/org/antennapod/audio/SonicAudioPlayer.java

```diff
@@ -516,53 +528,78 @@ private int findFormatFromChannels(int numChannels) {
         }
     }
 
-    public void initStream() throws IOException {
-        mLock.lock();
+    private String currentPath() {
+        if (mPath != null) {
+            return mPath;
+        }
+        else if (mUri != null) {
+            return mUri.toString();
+        }
+
+        return null;
+    }
+
+    public boolean initStream() throws IOException {
         mExtractor = new MediaExtractor();
+
+        String lastPath = currentPath();
```

Comment:
Thanks for your replay @mfietz! Do you mean, comment in the source code? Yes, I can.

Since we can't cancel an Media Extractor after setting the datasource, this is a simple (but not robust) way to detect if any raised error are related to the current episode.

Without the lock, when the error raises, there's a chance that the user already changed to another episode.